src/sentry/models/apitoken.py

```diff
@@ -252,6 +267,64 @@ def update(self, *args: Any, **kwargs: Any) -> int:
     def outbox_region_names(self) -> Collection[str]:
         return list(find_all_region_names())
 
+    def _should_flush_outbox(self) -> bool:
+        from sentry import options
+
+        has_async_flush = self.user_id in options.get("users:api-token-async-flush")
+        logger.info(
+            "async_flush_check",
+            extra={
+                "has_async_flush": has_async_flush,
+                "user_id": self.user_id,
+                "token_id": self.id,
+            },
+        )
+        if has_async_flush:
+            return False
+
+        return True
+
+    @contextlib.contextmanager
+    def _maybe_prepare_outboxes(self, *, outbox_before_super: bool) -> Generator[None]:
+        # Overriding to get around how default_flush cannot be cleanly feature flagged
+        flush = self._should_flush_outbox()
+
+        with outbox_context(
+            transaction.atomic(router.db_for_write(type(self))),
+            flush=flush,
+        ):
+            if not outbox_before_super:
+                yield
+            for outbox in self.outboxes_for_update():
+                outbox.save()
+            if outbox_before_super:
+                yield
+
+    def _schedule_async_replication(self) -> None:
```

Comment:
I chose to use `drain_outbox_shards_control` directly as `enqueue_outbox_jobs_control` didn't have as much granularity/control on what rows we're processing. I'm not 100 on if im using `drain_outbox_shards_control` correctly lol (and tbh there weren't like any controls for specifying X category for processing/flushing only).

Anyway what this is _supposed_ to do is find the outbox we just created and if there are any other outboxes on the same resource. Then enqueue `drain_outbox_shards_control` to process those rows (only?).